### ISSUE TITLE
Issue with Vercel Domain Creation

### ISSUE THREAD

**Issue by ianconsolata** [chronark/terraform-provider-vercel#12]
Hi there,

Thank you so much for making this module! I'll probably be making heavy use of it over the next couple months, and I really appreciate the work you put in. Unfortunately I couldn't get the new vercel domain stuff you just added to work quick right for me. Maybe I misconfigured something? Any help would be greatly appreciated.

### Terraform Version
```
Terraform v0.15.3
on darwin_amd64
+ provider registry.terraform.io/chronark/vercel v0.10.0
+ provider registry.terraform.io/hashicorp/github v4.9.4
+ provider registry.terraform.io/integrations/github v4.9.4
```

### Affected Resource(s)
Please list the resources as a list, for example:
- vercel_domain

### Terraform Configuration Files
```hcl
resource "vercel_domain" "antarahaldar_com" {
  team_id = var.vercel_team_id
  name = "antarahaldar.com"
}
```

### Debug Output
```
│ Error: Unable to fetch domain from vercel: Unable to request resource: [GET] /v4/domains/antarahaldar.com/?teamId=team_Mb1ivhnQAH2uo2nNrPBbDwk4 with payload {<nil>}: Error during http request: map[error:map[code:not_found message:The requested API endpoint was not found.]]
│
│   with vercel_domain.antarahaldar_com,
│   on antara.tf line 22, in resource "vercel_domain" "antarahaldar_com":
│   22: resource "vercel_domain" "antarahaldar_com" {
```

### Expected Behavior
It should create the domain and not error

### Actual Behavior
The domain is created as expected in vercel, but terraform thinks it failed and exists with this error.

**Comment by chronark** [comment]
Hey @ianconsolata thank you.

Sorry for being so late, my studies kept me from working on this. 
I'll have a deeper look in a few hours

**Comment by chronark** [comment]
I fixed it in `v10.0.1`, can you try that please

**Comment by dddaaammmooo** [comment]
@chronark I dont think this is working properly still--- at least in my testing the presence of a trailing slash prior to the `?teamId=xxx` parameter causes the "The requested API endpoint was not found" error- see attached image of curl request.

![image](https://user-images.githubusercontent.com/4394695/128115834-a3ab1b6a-e24e-41c4-b792-0688e10a39c0.png)

**Comment by chronark** [comment]
@dddaaammmooo  this should be fixed by your PR last week, shouldn't it?

If not: I am currently rewriting the go vercel api adapter [here](https://github.com/chronark/vercel-go) and it will fix itself once I use that in the terraform provider
